src/lib389/lib389/__init__.py

```diff
@@ -1090,6 +1090,7 @@ def start(self, timeout=120, post_open=True):
                 self.log.info("INFO: ASAN options will be copied from your environment")
                 env['ASAN_SYMBOLIZER_PATH'] = "/usr/bin/llvm-symbolizer"
                 env['ASAN_OPTIONS'] = "symbolize=1 detect_deadlocks=1 log_path=%s/ns-slapd-%s.asan" % (self.ds_paths.run_dir, self.serverid)
+                env['LD_BIND_NOW'] = "yes"
```

Comment:
Have you tried to use  LD_PRELOAD=libcrypt.so instead of LD_BIND_NOW=yes ?

My concern is that LD_BIND_NOW has a noticeable impact on ns-slapd start time 
  that will be painful when running all the pytests to check for non regression.
 ( FYI: on my machine when using LD_BIND_NOW=yes start time increased by 2.5 seconds. )
That said, IMHO if we do not find any better solution, LD_BIND_NOW is still acceptable.

Reply:
At a 2.5 second increase per restart it would increase the testing time by at least 1.5 to 2 hours, so if there are other options like LD_PRELOAD they should be explored.

Reply:
I can remove the LD_BIND_NOW, I think the change to linking in libslapd is actually enough to resolve the issue.